### PR TITLE
More specific types for ReactTag

### PR DESCRIPTION
**Pull request by elacin**
PR for #199

### CODE REVIEW

**Comment by elacin** [comment]
The `RouterCtl` change is not needed, but i figured it wouldnt hurt to do the same there too. Didn't have to pull that invariant trick since this change doesnt include the ref changes i proposed

File: core/src/main/scala/japgolly/scalajs/react/vdom/package.scala [hunk 1]
```diff
@@ -19,7 +19,7 @@ package object vdom {
 
   // If you're wondering why abstract class instead of trait, https://issues.scala-lang.org/browse/SI-4767
   abstract class Base extends Implicits {
-    final type ReactTag = japgolly.scalajs.react.vdom.ReactTag
+    final type ReactTag = japgolly.scalajs.react.vdom.ReactTagOf[TopNode]
```
Comment (japgolly) [inline core/src/main/scala/japgolly/scalajs/react/vdom/package.scala]: We should probably add a new type alias here for `ReactTagOf`

**Comment by japgolly** [comment]
Only one small comment above. Otherwise, looks good. Would you also mind just making a short note of this towards the bottom of the changelog?

File: core/src/main/scala/japgolly/scalajs/react/vdom/package.scala (hunk 1 above)
Comment (elacin) [inline core/src/main/scala/japgolly/scalajs/react/vdom/package.scala]: Ah yes, of course.

**Comment by elacin** [comment]
Will fix both things today, just a day of work first :)

**Comment by japgolly** [comment]
Looks great. Love your work as usual. In it goes; thanks!

**Pull request merged**